### PR TITLE
Implement Spark's version of log2, log10

### PR DESCRIPTION
**Pull request by zhztheplayer**

### CODE REVIEW

File: velox/functions/prestosql/Arithmetic.h [hunk 1]
```diff
@@ -198,6 +198,30 @@ struct Log10Function {
   }
 };
 
+template <typename T>
+struct Log2FunctionNaNAsNull {
+  FOLLY_ALWAYS_INLINE bool call(double& result, double a) {
+    double yAsymptote = 0.0;
+    if (a <= yAsymptote) {
```
Comment (rui-mo) [inline velox/functions/prestosql/Arithmetic.h:205]: Use std::isnan for NaN?

Comment (zhztheplayer) [reply velox/functions/prestosql/Arithmetic.h:205]: Did you mean using `std::isnan(a)` to replace `a <= yAsymptote` ? The two don't seem to share the same semantic?

Comment (zhztheplayer) [reply velox/functions/prestosql/Arithmetic.h:205]: btw NaN is the output of log(-1) for example.

Comment (rui-mo) [reply velox/functions/prestosql/Arithmetic.h:205]: Got it. With above logic, will `null` be returned for this case instead of `NaN`?

Comment (zhztheplayer) [reply velox/functions/prestosql/Arithmetic.h:205]: Yes, Velox treats the return type to be nullable if `bool call(...)` is used here.

File: velox/functions/prestosql/Arithmetic.h [hunk 2]
```diff
@@ -198,6 +198,30 @@ struct Log10Function {
   }
 };
 
+template <typename T>
```
Comment (jinchengchenghh) [inline velox/functions/prestosql/Arithmetic.h:201]: why register the function in sparksql folder and implement it in prestosql folder

**Pull request merged**